### PR TITLE
1819 document featsql udf support named sql udf

### LINKED ISSUE #1819: Document: feat(sql-udf): support named sql udf
This issue tracks the documentation update needed for the merged PR #14806.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/14806
Source PR Merged At: 2024-01-30T20:18:40Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [ Add a mock table example to anonymous sql udfs]
  - [Add many examples to named sql udfs ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [https://github.com/risingwavelabs/risingwave/pull/14806/files ]

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1819#event-11694410609

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Comment by xzhseh** [comment]
After some discussion with @xxchan in https://github.com/risingwavelabs/risingwave/pull/14941#discussion_r1475606146, let's change the current description of **[anonymous / named sql udf]** to **[sql udf with unnamed / named parameters]**, which should then be clear and concise in most cases.

Thanks in advance!

P.S. Since the test cases (i.e., the `sql_udf.slt`) are updated and refactored, it should be clearer to find more useful examples at present, if needed.

**Comment by hengm3467** [comment]
@xzhseh Could you please take a look again? Thanks!

File: docs/sql/commands/sql-create-function.md [hunk 1]
```diff
@@ -126,22 +137,85 @@ select sub(1, 1);
 0
 ```
 
-- `AS` clause with dollar definition
+- Create a SQL UDF with unnamed parameters that calls other pre-defined SQL UDFs.
 
 ```sql title="Create function"
-create function add(INT, INT) returns int language sql as $$select $1 + $2$$;
+-- Create two pre-defined SQL UDFs
```
Comment (xzhseh) [inline docs/sql/commands/sql-create-function.md]:
```suggestion
-- Create two SQL UDFs for subsequent call
```

File: docs/sql/commands/sql-create-function.md [hunk 2]
```diff
@@ -150,74 +224,231 @@ select add_return(1, 1);
...
 ```sql title="Call function"
 select add_sub(1, 5.1415926, 1);
 ----RESULT
 3.1415926
+```
 
-select add_sub_return(1, 5.1415926, 1);
-----RESULT
-3.1415926
+- Create a SQL UDF with complex types of parameters.
```
Comment (xzhseh) [inline docs/sql/commands/sql-create-function.md]:
```suggestion
- Create a SQL UDF with complex types of unnamed parameters.
```

File: docs/sql/commands/sql-create-function.md [hunk 3]
```diff
@@ -150,74 +224,231 @@ select add_return(1, 1);
 2
 ```
 
-- SQL UDF with input of different data types
+- Create a SQL UDF with a return expression using previously defined UDFs.
 
 ```sql title="Create function"
--- Multiple type interleaving
-create function add_sub(INT, FLOAT, INT) returns float language sql as $$select -$1 + $2 - $3$$;
+-- Create a pre-defined UDF
```
Comment (xzhseh) [inline docs/sql/commands/sql-create-function.md]:
```suggestion
-- Create a SQL UDF used in subsequent call
```

File: docs/sql/commands/sql-create-function.md [hunk 4]
```diff
@@ -150,74 +224,231 @@ select add_return(1, 1);
...
+
+```sql title="Call function"
+select add_sub_wrapper(1, 1);
+----RESULT
+114514
+```
+
+#### Basic SQL UDFs integrated with the use of mock tables 
+
+The examples in this section are a simulation of real-world use cases.
```
Comment (xzhseh) [inline docs/sql/commands/sql-create-function.md]:
```suggestion
The examples in this section are a basic simulation of real-world use cases.
```

File: docs/sql/commands/sql-create-function.md [hunk 5]
```diff
@@ -150,74 +224,231 @@ select add_return(1, 1);
...
+1 3.14 2 3 -1 0.14000000000000012
+2 4.44 5 7 -3 -2.5599999999999996
+20 10.3 2 22 18 -11.7
+
+select c1, c2, c3, add_mc(c1, c3), sub_mc(c1, c3), add_sub_return_mc(c1, c2, c3) from t2 order by c1 asc;
+----RESULT
+1 3.14 2 3 -1 0.14000000000000012
+2 4.44 5 7 -3 -2.5599999999999996
+20 10.3 2 22 18 -11.7
+
```
Comment (xzhseh) [inline docs/sql/commands/sql-create-function.md]: Remove the empty line here

File: docs/sql/commands/sql-create-function.md [hunk 6]
```diff
@@ -150,74 +224,231 @@ select add_return(1, 1);
...
+```sql title="Call function"
+select corner_case(1, 2, 3);
+----RESULT
+$1 + a + $3
+```
+
+---
+
+
```
Comment (xzhseh) [inline docs/sql/commands/sql-create-function.md]: ditto

**Review by xzhseh — COMMENTED** [review]
Rest LGTM, thanks for the effort!

**Review by CharlieSYH — APPROVED** [review]
LGTM!

**Pull request merged**